velox/exec/Window.cpp

```diff
@@ -345,8 +511,54 @@ void Window::callApplyForPartitionRows(
     // as WindowFunction only sees one partition at a time.
     rawPeerStarts[j] = peerStartRow_ - firstPartitionRow;
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
+    if (precedingFlatVector) {
+      startOffset = precedingFlatVector->valueAt(j);
```

Comment:
Maybe you should initialize precedingFlatVector as a constant vector of value 0 and always get startOffset from that vector instead of doing a check for the pointer each time.

Reply:
Thanks for the suggestion, will incorporate this in the PR for variable k values; removed precedingVector from this PR.